src/main/java/com/puppycrawl/tools/checkstyle/xpath/AbstractNode.java

```diff
@@ -0,0 +1,522 @@
+import net.sf.saxon.om.NodeInfo;
+import net.sf.saxon.om.SequenceIterator;
+import net.sf.saxon.om.TreeInfo;
+import net.sf.saxon.pattern.NodeTest;
+import net.sf.saxon.trans.XPathException;
+import net.sf.saxon.tree.iter.AxisIterator;
+import net.sf.saxon.tree.util.FastStringBuffer;
+import net.sf.saxon.tree.util.Navigator;
+import net.sf.saxon.type.SchemaType;
+
+/**
+ * Represents general class for {@code ElementNode}, {@code RootNode} and {@code AttributeNode}.
+ *
+ * @author Timur Tibeyev
+ */
+// -@cs[MethodCount] This class implements all interface methods. Total number of methods is 47.
+// No way to decrease total number of methods, because interface is the part of the Saxon library.
+public abstract class AbstractNode implements NodeInfo {
+    /** The map of {@link TokenTypes} to {@code String}. */
+    private static final Map<Integer, String> TOKEN_TYPES_TO_STRING_MAP = new HashMap<>();
+
+    /** The children. */
+    private final List<AbstractNode> children = new ArrayList<>();
+
+    /** The token type. */
+    private int tokenType;
+
+    /** The line number. */
+    private int lineNo;
+
+    /** The column number. */
+    private int columnNo;
+
+    /**
+     * Getter method for token type.
+     * @return token type
+     */
+    public int getTokenType() {
+        return tokenType;
+    }
+
+    /**
+     * Setter method for token type.
+     * @param tokenType token type
+     */
+    protected void setTokenType(int tokenType) {
+        this.tokenType = tokenType;
+    }
+
+    /**
+     * Setter method for line number.
+     * @param lineNo line number
+     */
+    protected void setLineNo(int lineNo) {
+        this.lineNo = lineNo;
+    }
+
+    /**
+     * Setter method for column number.
+     * @param columnNo column number
+     */
+    protected void setColumnNo(int columnNo) {
+        this.columnNo = columnNo;
+    }
+
+    /**
+     * Getter method for children.
+     * @return children list
+     */
+    protected List<AbstractNode> getChildren() {
+        return Collections.unmodifiableList(children);
+    }
+
+    /**
+     * Add new child node to children list.
+     * @param node child node
+     */
+    protected void addChild(AbstractNode node) {
+        children.add(node);
+    }
+
+    /**
+     * Returns true if nodes are same, false otherwise.
+     * @param nodeInfo other node
+     * @return {@code TreeInfo}
+     */
+    @Override
+    public boolean isSameNodeInfo(NodeInfo nodeInfo) {
+        return this == nodeInfo;
+    }
+
+    /**
+     * Returns if implementation provides fingerprints.
+     * @return {@code boolean}
+     */
+    @Override
+    public boolean hasFingerprint() {
+        return false;
+    }
+
+    /**
+     * Returns uri of the namespace for the current node.
+     * @return uri
+     */
+    // -@cs[AbbreviationAsWordInName] Implements method from the interface.
+    // No way to change method name, because interface is the part of the Saxon library.
```

Comment:
@MEZk ,

> Thus, from my point of view we should use ignoreOverriddenMethods option of the check.

good, please next time, be skeptical on each attempt to suppress, it should be always very good reason to suppress and ok to suppress if case is very special and very rare. In all other case we need to think on how to make Check better and raise less false-positives to user.

this option is enabled by default - http://checkstyle.sourceforge.net/config_naming.html#AbbreviationAsWordInName , do we really have there violation ? is it bug in Check ?